File: Bwl.Imaging/Geometry/Structures/Segment.vb
Imports System.Drawing

Public Class Segment
    Public Property Left As Integer
    Public Property Top As Integer
    Public Property Width As Integer
    Public Property Height As Integer
    Public Property ID As Integer
    Public Property Tag As Integer
    Public Property Debug As String

    Public Property Right As Integer
        Get
            Return Left + Width
        End Get
        Set(value As Integer)
            Width = value - Left
        End Set
    End Property
    Public Property Bottom As Integer
        Get
            Return Top + Height
        End Get
        Set(value As Integer)
            Height = value - Top
        End Set
    End Property
    Public Overrides Function ToString() As String
        Return "L:" + Left.ToString + " :T" + Top.ToString + " :W" + Width.ToString + " :H" + Height.ToString
    End Function

    Public ReadOnly Property CenterX As Integer
        Get
            Return CInt(Left + Width / 2)
        End Get
    End Property

    Public ReadOnly Property WHRatio As Single
        Get
            If Height = 0 Then Return 0
            Return CInt(Width / Height)
        End Get
    End Property

    Public ReadOnly Property CenterY As Integer
        Get
            Return CInt(Top + Height / 2)
        End Get
    End Property

    Public Function IsPointInside(x As Integer, y As Integer) As Boolean
        Return x >= Left And x <= Left + Width And y >= Top And y <= Top + Height
    End Function

    Public Function ToRectangle() As Rectangle
        Return New Rectangle(Left, Top, Width, Height)
    End Function

    Public Function ToRectangleF() As RectangleF
        Return New RectangleF(Left, Top, Width, Height)
    End Function

    Public Sub New()

    End Sub

    Public Sub New(x As Integer, y As Integer, width As Integer, height As Integer)
        Me.Left = x
        Me.Top = y
        Me.Width = width
        Me.Height = height
    End Sub

End Class

File: Bwl.Imaging/Geometry/Structures/Segment.cs
using System;
using System.Drawing;

namespace Bwl.Imaging
{

    public class Segment
    {
        public int Left { get; set; }
        public int Top { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public int ID { get; set; }
        public int Tag { get; set; }
        public string Debug { get; set; }

        public int Right
        {
            get
            {
                return Left + Width;
            }
            set
            {
                Width = value - Left;
            }
        }
        public int Bottom
        {
            get
            {
                return Top + Height;
            }
            set
            {
                Height = value - Top;
            }
        }
        public override string ToString()
        {
            return "L:" + Left.ToString() + " :T" + Top.ToString() + " :W" + Width.ToString() + " :H" + Height.ToString();
        }

        public int CenterX
        {
            get
            {
                return (int)Math.Round(Left + Width / 2d);
            }
        }

        public float WHRatio
        {
            get
            {
                if (Height == 0)
                    return 0f;
                return (int)Math.Round(Width / (double)Height);
            }
        }

        public int CenterY
        {
            get
            {
                return (int)Math.Round(Top + Height / 2d);
            }
        }

        public bool IsPointInside(int x, int y)
        {
            return x >= Left & x <= Left + Width & y >= Top & y <= Top + Height;
        }

        public Rectangle ToRectangle()
        {
            return new Rectangle(Left, Top, Width, Height);
        }

        public RectangleF ToRectangleF()
        {
            return new RectangleF(Left, Top, Width, Height);
        }

        public Segment()
        {

        }

        public Segment(int x, int y, int width, int height)
        {
            Left = x;
            Top = y;
            Width = width;
            Height = height;
        }

    }
}